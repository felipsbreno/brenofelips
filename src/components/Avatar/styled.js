import styled from "styled-components"
import Image from "gatsby-image"

export const AvatarWrapper = styled(Image)`
  border-radius: 50%;
  height: 3.75rem;
  margin: auto;
  width: 3.75rem;
`
