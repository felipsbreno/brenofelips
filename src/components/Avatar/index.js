import React from "react"
import { StaticImage } from "gatsby-plugin-image"

import * as S from "./styled"

const Avatar = () => {
  return (
    <S.AvatarWrapper>
      <StaticImage
        src="../../images/my.jpg"
        alt="John Doe Avatar"
        placeholder="blurred"
      />
    </S.AvatarWrapper>
  )
}

export default Avatar
