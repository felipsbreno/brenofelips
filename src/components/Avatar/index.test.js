import React from "react"
import { screen, render } from "@testing-library/react"
import Avatar from "."

describe("Avatar", () => {
  it("should be equals snapshot", () => {
    const { container } = render(<Avatar />)
    expect(container).toMatchSnapshot()
  })

  it("should be render a component Avatar", () => {
    render(<Avatar />)

    const getElementById = screen.getByTestId("avatar")
    expect(getElementById).toBeInTheDocument()
  })
})
