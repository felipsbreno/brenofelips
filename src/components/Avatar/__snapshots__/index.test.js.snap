// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should be equals snapshot 1`] = `
.c0 .gatsby-image-wrapper {
  border-radius: 50%;
  height: 3.75rem;
  margin: auto;
  width: 3.75rem;
}

@media (max-width: 1170px) {
  .c0 .gatsby-image-wrapper {
    height: 1.875rem;
    width: 1.875rem;
  }
}

<div>
  <div
    class="c0"
    data-testid="avatar"
  />
</div>
`;
